src/gei/Commands/GenerateArchiveCommand.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class GenerateArchiveCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        public GenerateArchiveCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("generate-archive")
```

Comment:
Are we planning on having the archive generation be a separate step? Or have this whole GHES migration (archive generation, dowload/upload to azure blob, then kick off octoshift) be in one command? If the latter, should we update the name to something like `MigrateGHESRepoCommand`?

Also, if the latter, we'll have lots of overlap with the `MigrateRepoCommand`, maybe we can consolidate these in a followup PR

Reply:
I'm feeling MigrateGHESRepoCommand :+1:

Reply:
We talked online, and decided to address in follow up